cmd/argocd-commit-server/commands/argocd_commit_server.go

```diff
@@ -57,7 +66,17 @@ func NewCommand() *cobra.Command {
 			askPassServer := askpass.NewServer(askpass.CommitServerSocketPath)
 			go func() { errors.CheckError(askPassServer.Run()) }()
 
-			server := commitserver.NewServer(askPassServer, metricsServer)
+			config, err := clientConfig.ClientConfig()
+			errors.CheckError(err)
+			kubeclientset := kubernetes.NewForConfigOrDie(config)
+
+			settingsMgr := settings.NewSettingsManager(context.Background(), kubeclientset, namespace)
+			err = settingsMgr.ResyncInformers()
+			if err != nil {
+				errors.CheckError(err)
+			}
```

Comment:
I think instead of loading the template directly from the commit-server component, we should load it in the application-controller and pass it to the commit-server via the gRPC request. That'll require updating commit.proto with the new param.